Data.Text.show causes build error.

Since text >= 2.1, A new function Data.Text.show is included, and it'll cause ambiguous with Prelude.show.

Mainly `Send` methods in telegram-bot-api are affected.

```
[174 of 225] Compiling Telegram.Bot.API.Methods.SendAudio ( src/Telegram/Bot/API/Methods/SendAudio.hs, dist/build/Telegram/Bot/API/Methods/SendAudio.o, dist/build/Telegram/Bot/API/Methods/SendAudio.dyn_o )
src/Telegram/Bot/API/Methods/SendAudio.hs:69:51: error: [GHC-87543]
    Ambiguous occurrence ‘show’.
    It could refer to
       either ‘Prelude.show’,
              imported from ‘Prelude’ at src/Telegram/Bot/API/Methods/SendAudio.hs:11:8-41
              (and originally defined in ‘ghc-internal-9.1202.0:GHC.Internal.Show’),
           or ‘Data.Text.show’,
              imported from ‘Data.Text’ at src/Telegram/Bot/API/Methods/SendAudio.hs:20:1-16.
   |
69 |           SomeChatId (ChatId chat_id) -> T.pack $ show chat_id
   |                                                   ^^^^
```

--- a/telegram-bot-api/src/Telegram/Bot/API/Internal/Utils.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Internal/Utils.hs
@@ -13,6 +13,7 @@ import Data.Aeson.TH (deriveJSON)
 import Data.Aeson.Types (Options(..), defaultOptions, Parser, Pair)
 import Data.Char (isUpper, toUpper, toLower)
 import Data.List (intercalate)
+import Data.Text (Text)
 import GHC.Generics
 import Language.Haskell.TH
 import Servant.Multipart.API (MultipartData(MultipartData), Input)
@@ -23,6 +24,7 @@ import qualified Data.Aeson.KeyMap as Map
 #else
 import qualified Data.HashMap.Strict as Map
 #endif
+import qualified Data.Text as T
 
 deriveJSON' :: Name -> Q [Dec]
 deriveJSON' name = deriveJSON (jsonOptions (nameBase name)) name
@@ -115,3 +117,11 @@ isNull _ = False
 addMultipartFields :: [Input] -> MultipartData tag -> MultipartData tag
 addMultipartFields newFields (MultipartData currenFields files)
       = MultipartData (newFields <> currenFields) files
+
+showText :: Show a => a -> Text
+#if MIN_VERSION_text(2,1,2)
+showText = T.show
+#else
+showText = T.pack . show
+#endif
+{-# INLINE showText #-}

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendAnimation.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendAnimation.hs
@@ -68,11 +68,11 @@ instance ToMultipart Tmp SendAnimationRequest where
     MultipartData fields [] where
     fields =
       [ Input "chat_id" $ case sendAnimationChatId of
-          SomeChatId (ChatId chat_id) -> T.pack $ show chat_id
+          SomeChatId (ChatId chat_id) -> showText chat_id
           SomeChatUsername txt -> txt
       ] <> catMaybes
       [ sendAnimationMessageThreadId <&>
-        \t -> Input "message_thread_id" (T.pack $ show t)
+        \t -> Input "message_thread_id" (showText t)
       , sendAnimationCaption <&>
         \t -> Input "caption" t
       , sendAnimationParseMode <&>

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendAnimation.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendAnimation.hs
@@ -24,7 +24,6 @@ import Servant.Multipart.API
 import Servant.Multipart.Client
 import Servant.Client hiding (Response)
 
-import qualified Data.Text as T
 import qualified Data.Text.Lazy as TL
 
 import Telegram.Bot.API.Internal.Utils

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendAudio.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendAudio.hs
@@ -66,11 +66,11 @@ instance ToMultipart Tmp SendAudioRequest where
     MultipartData fields [] where
     fields =
       [ Input "chat_id" $ case sendAudioChatId of
-          SomeChatId (ChatId chat_id) -> T.pack $ show chat_id
+          SomeChatId (ChatId chat_id) -> showText chat_id
           SomeChatUsername txt -> txt
       ] <> catMaybes
       [ sendAudioMessageThreadId <&>
-        \t -> Input "message_thread_id" (T.pack $ show t)
+        \t -> Input "message_thread_id" (showText t)
       , sendAudioCaption <&>
         \t -> Input "caption" t
       , sendAudioParseMode <&>

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendAudio.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendAudio.hs
@@ -24,7 +24,6 @@ import Servant.Multipart.API
 import Servant.Multipart.Client
 import Servant.Client hiding (Response)
 
-import qualified Data.Text as T
 import qualified Data.Text.Lazy as TL
 
 import Telegram.Bot.API.Internal.Utils

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendDocument.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendDocument.hs
@@ -96,10 +96,10 @@ instance ToMultipart Tmp SendDocumentRequest where
     fields =
       [ Input "document" "attach://file"
       , Input "chat_id" $ case sendDocumentChatId of
-          SomeChatId (ChatId chat_id) -> T.pack $ show chat_id
+          SomeChatId (ChatId chat_id) -> showText chat_id
           SomeChatUsername txt -> txt
       ] <>
-          maybe id (\t -> (Input "message_thread_id" (T.pack $ show t) :)) sendDocumentMessageThreadId
+          maybe id (\t -> (Input "message_thread_id" (showText t) :)) sendDocumentMessageThreadId
         ( maybe id (const (Input "thumbnail" "attach://thumbnail" :)) sendDocumentThumbnail
         $ maybe id (\t -> (Input "caption" t :)) sendDocumentCaption
         $ maybe id (\t -> (Input "parse_mode" (TL.toStrict $ encodeToLazyText t) :)) sendDocumentParseMode

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendPhoto.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendPhoto.hs
@@ -85,10 +85,10 @@ instance ToMultipart Tmp SendPhotoRequest where
     fields =
       [ Input "photo" "attach://file"
       , Input "chat_id" $ case sendPhotoChatId of
-          SomeChatId (ChatId chat_id) -> T.pack $ show chat_id
+          SomeChatId (ChatId chat_id) -> showText chat_id
           SomeChatUsername txt -> txt
       ] <>
-          maybe id (\t -> (Input "message_thread_id" (T.pack $ show t) :)) sendPhotoMessageThreadId
+          maybe id (\t -> (Input "message_thread_id" (showText t) :)) sendPhotoMessageThreadId
         ( maybe id (const (Input "thumb" "attach://thumb" :)) sendPhotoThumb
         $ maybe id (\t -> (Input "caption" t :)) sendPhotoCaption
         $ maybe id (\t -> (Input "parse_mode" (TL.toStrict . TL.replace "\"" "" $ encodeToLazyText t) :)) sendPhotoParseMode

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVideo.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVideo.hs
@@ -24,7 +24,6 @@ import Servant.Multipart.API
 import Servant.Multipart.Client
 import Servant.Client hiding (Response)
 
-import qualified Data.Text as T
 import qualified Data.Text.Lazy as TL
 
 import Telegram.Bot.API.Internal.Utils

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVideo.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVideo.hs
@@ -69,11 +69,11 @@ instance ToMultipart Tmp SendVideoRequest where
     MultipartData fields [] where
     fields =
       [ Input "chat_id" $ case sendVideoChatId of
-          SomeChatId (ChatId chat_id) -> T.pack $ show chat_id
+          SomeChatId (ChatId chat_id) -> showText chat_id
           SomeChatUsername txt -> txt
       ] <> catMaybes
       [ sendVideoMessageThreadId <&>
-        \t -> Input "message_thread_id" (T.pack $ show t)
+        \t -> Input "message_thread_id" (showText t)
       , sendVideoCaption <&>
         \t -> Input "caption" t
       , sendVideoParseMode <&>

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVideoNote.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVideoNote.hs
@@ -24,7 +24,6 @@ import Servant.Multipart.API
 import Servant.Multipart.Client
 import Servant.Client hiding (Response)
 
-import qualified Data.Text as T
 import qualified Data.Text.Lazy as TL
 
 import Telegram.Bot.API.Internal.Utils

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVideoNote.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVideoNote.hs
@@ -61,11 +61,11 @@ instance ToMultipart Tmp SendVideoNoteRequest where
     MultipartData fields [] where
     fields =
       [ Input "chat_id" $ case sendVideoNoteChatId of
-          SomeChatId (ChatId chat_id) -> T.pack $ show chat_id
+          SomeChatId (ChatId chat_id) -> showText chat_id
           SomeChatUsername txt -> txt
       ] <> catMaybes
       [ sendVideoNoteMessageThreadId <&>
-        \t -> Input "message_thread_id" (T.pack $ show t)
+        \t -> Input "message_thread_id" (showText t)
       , sendVideoNoteDisableNotification <&>
         \t -> Input "disable_notification" (bool "false" "true" t)
       , sendVideoNoteProtectContent <&>

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVoice.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVoice.hs
@@ -24,7 +24,6 @@ import Servant.Multipart.API
 import Servant.Multipart.Client
 import Servant.Client hiding (Response)
 
-import qualified Data.Text as T
 import qualified Data.Text.Lazy as TL
 
 import Telegram.Bot.API.Internal.Utils

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVoice.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SendVoice.hs
@@ -62,11 +62,11 @@ instance ToMultipart Tmp SendVoiceRequest where
     MultipartData fields [] where
     fields =
       [ Input "chat_id" $ case sendVoiceChatId of
-          SomeChatId (ChatId chat_id) -> T.pack $ show chat_id
+          SomeChatId (ChatId chat_id) -> showText chat_id
           SomeChatUsername txt -> txt
       ] <> catMaybes
       [ sendVoiceMessageThreadId <&>
-        \t -> Input "message_thread_id" (T.pack $ show t)
+        \t -> Input "message_thread_id" (showText t)
       , sendVoiceCaption <&>
         \t -> Input "caption" t
       , sendVoiceParseMode <&>

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SetChatPhoto.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SetChatPhoto.hs
@@ -19,6 +19,7 @@ import qualified Data.Text as T
 import Telegram.Bot.API.MakingRequests
 import Telegram.Bot.API.Types
 import Telegram.Bot.API.Internal.TH
+import Telegram.Bot.API.Internal.Utils (showText)
 
 -- ** 'setChatPhoto'
 
@@ -33,7 +34,7 @@ instance ToMultipart Tmp SetChatPhotoRequest where
     makeFile "photo" setChatPhotoPhoto (MultipartData fields []) where
     fields =
       [ Input "chat_id" $ case setChatPhotoChatId of
-          SomeChatId (ChatId chat_id) -> T.pack $ show chat_id
+          SomeChatId (ChatId chat_id) -> showText chat_id
           SomeChatUsername txt -> txt
       ]
 

--- a/telegram-bot-api/src/Telegram/Bot/API/Methods/SetChatPhoto.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Methods/SetChatPhoto.hs
@@ -14,8 +14,6 @@ import Servant.Multipart.API
 import Servant.Multipart.Client
 import Servant.Client hiding (Response)
 
-import qualified Data.Text as T
-
 import Telegram.Bot.API.MakingRequests
 import Telegram.Bot.API.Types
 import Telegram.Bot.API.Internal.TH

--- a/telegram-bot-api/src/Telegram/Bot/API/Stickers.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Stickers.hs
@@ -13,6 +13,8 @@ import Control.Monad.IO.Class
 import Data.Aeson
 import Data.Aeson.Text
 import Data.Bool
+import Data.Functor
+import Data.Maybe (catMaybes)
 import Data.Text (Text)
 import qualified Data.Text as T
 import qualified Data.Text.Lazy as TL
@@ -23,13 +25,10 @@ import Servant.Client hiding (Response)
 import Servant.Multipart.API
 import Servant.Multipart.Client
 
+import Telegram.Bot.API.Internal.TH (makeDefault)
 import Telegram.Bot.API.Internal.Utils
 import Telegram.Bot.API.MakingRequests (Response)
 import Telegram.Bot.API.Types
-import Data.Maybe (catMaybes)
-import Data.Functor
-import Telegram.Bot.API.Internal.TH (makeDefault)
-
 
 -- | Type of uploaded sticker file. Static or animated.
 data StickerType
@@ -66,11 +65,11 @@ instance ToMultipart Tmp SendStickerRequest where
     makeFile "sticker" sendStickerSticker (MultipartData fields []) where
     fields =
       [ Input "chat_id" $ case sendStickerChatId of
-          SomeChatId (ChatId chat_id) -> T.pack $ show chat_id
+          SomeChatId (ChatId chat_id) -> showText chat_id
           SomeChatUsername txt -> txt
       ] <> catMaybes
       [ sendStickerMessageThreadId <&>
-        \t -> Input "message_thread_id" (T.pack $ show t)
+        \t -> Input "message_thread_id" (showText t)
       , sendStickerDisableNotification <&>
         \t -> Input "disable_notification" (bool "false" "true" t)
       , sendStickerProtectContent <&>

--- a/telegram-bot-api/src/Telegram/Bot/API/Webhook.hs
+++ b/telegram-bot-api/src/Telegram/Bot/API/Webhook.hs
@@ -32,7 +32,7 @@ import           Servant.Client                      (ClientEnv, ClientError,
 import           Servant.Multipart.API
 import           Servant.Multipart.Client            (genBoundary)
 
-import           Telegram.Bot.API.Internal.Utils     (gtoJSON)
+import           Telegram.Bot.API.Internal.Utils     (gtoJSON, showText)
 import           Telegram.Bot.API.MakingRequests     (Response)
 import           Telegram.Bot.API.Types              (InputFile, makeFile)
 import Telegram.Bot.API.Internal.TH (makeDefault)
@@ -67,7 +67,7 @@ instance ToMultipart Tmp SetWebhookRequest where
           <> catMaybes
             [ setWebhookSecretToken <&> \t -> Input "secret_token" $ Text.pack t,
               setWebhookIpAddress <&> \t -> Input "ip_address" $ Text.pack t,
-              setWebhookMaxConnections <&> \t -> Input "max_connections" $ Text.pack $ show t,
+              setWebhookMaxConnections <&> \t -> Input "max_connections" $ showText t,
               setWebhookDropPendingUpdates <&> \t -> Input "drop_pending_updates" (bool "false" "true" t),
               setWebhookAllowedUpdates <&> \t -> Input "allowed_updates" (arrToJson t)
             ]

